whisper/whisperv5/whisper.go

```diff
@@ -673,17 +749,24 @@ func (w *Whisper) expire() {
 }
 
 // Stats returns the whisper node statistics.
-func (w *Whisper) Stats() string {
-	result := fmt.Sprintf("Memory usage: %d bytes. Average messages cleared per expiry cycle: %d. Total messages cleared: %d.",
-		w.stats.memoryUsed, w.stats.totalMessagesCleared/w.stats.cycles, w.stats.totalMessagesCleared)
-	if w.stats.messagesCleared > 0 {
-		result += fmt.Sprintf(" Latest expiry cycle cleared %d messages (%d bytes).",
-			w.stats.messagesCleared, w.stats.memoryCleared)
-	}
-	if w.overflow {
-		result += " Message queue state: overflow."
-	}
-	return result
+func (w *Whisper) Stats() Statistics {
+	w.statsMu.Lock()
+	defer w.statsMu.Unlock()
+
+	return w.stats
```

Comment:
The locking here seems unnecessary, you're returning the `w.stats` object, so if the caller does modifications on it, these locks won't prevent that.